kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -607,6 +607,44 @@ class DefaultExpressionEvaluatorSuite extends AnyFunSuite with ExpressionSuiteBa
       ),
       (ofDate(-12123), ofDate(123123), ofDate(-12123), ofNull(DateType.DATE)),
       (ofString("apples"), ofString("oranges"), ofString("apples"), ofNull(StringType.STRING)),
+      (
+        ofString("apples"),
+        ofString("oranges"),
+        ofString("apples"),
+        ofNull(StringType.STRING)
+      ),
+      (
+        ofString("abc"),
+        ofString("abcd"),
+        ofString("abc"),
+        ofNull(StringType.STRING)
+      ),
+      (
+        ofString("abc"),
+        ofString("abd"),
+        ofString("abc"),
+        ofNull(StringType.STRING)
+      ),
+      (
+        // scalastyle:off nonascii
+        ofString("A"),
+        ofString("\u0100"),
+        ofString("A"),
+        ofNull(StringType.STRING)
+      ),
+      (
+        ofString("\00BB"),
+        ofString("\u00EE"),
+        ofString("\00BB"),
+        ofNull(StringType.STRING)
+      ),
+      (
+        ofString("\uFFFD"),
+        ofString("\uD83C\uDF3C"),
+        ofString("\uFFFD"),
+        ofNull(StringType.STRING)
+        // scalastyle:on nonascii
```

Comment:
are there more tests we could copy for Spark or Delta-Spark around other surrogate pairs and unsigned char comparison?